pyte/screens.py

```diff
@@ -450,7 +450,8 @@ def draw(self, data):
 
             line = self.buffer[self.cursor.y]
             if char_width == 1:
-                line[self.cursor.x] = self.cursor.attrs._replace(data=char)
+                if char != self.cursor.attrs.data:
```

Comment:
Yes, I rolled that back. 

Tried idea with defaultchar, and it does not give gain in performance